From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xose Vazquez Perez <xose.vazquez@gmail.com>
Date: Sat, 5 Jun 2021 01:01:45 +0200
Subject: [PATCH] multipath-tools: add info about HPE Alletra 6000 and 9000

Cc: Martin Wilck <mwilck@suse.com>
Cc: Benjamin Marzinski <bmarzins@redhat.com>
Cc: Christophe Varoqui <christophe.varoqui@opensvc.com>
Cc: DM-DEVEL ML <dm-devel@redhat.com>
Signed-off-by: Xose Vazquez Perez <xose.vazquez@gmail.com>
Reviewed-by: Martin Wilck <mwilck@suse.com>
Signed-off-by: Benjamin Marzinski <bmarzins@redhat.com>
---
 README.alua            | 2 +-
 libmultipath/hwtable.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/README.alua b/README.alua
index b15eb487..5d2b1c64 100644
--- a/README.alua
+++ b/README.alua
@@ -6,7 +6,7 @@ To enable ALUA, the following options should be changed:
 - EMC CLARiiON/VNX:
    "Failover Mode" should be changed to "4" or "Active-Active mode(ALUA)-failover mode 4"
 
-- HPE 3PAR:
+- HPE 3PAR, Primera, and Alletra 9000:
    "Host:" should be changed to "Generic-ALUA Persona 2 (UARepLun, SESLun, ALUA)".
 
 - Promise VTrak/Vess:
diff --git a/libmultipath/hwtable.c b/libmultipath/hwtable.c
index 58fa7387..e884d8c7 100644
--- a/libmultipath/hwtable.c
+++ b/libmultipath/hwtable.c
@@ -107,7 +107,7 @@ static struct hwentry default_hw[] = {
 	 * HPE
 	 */
 	{
-		/* 3PAR / Primera */
+		/* 3PAR / Primera / Alletra 9000 */
 		.vendor        = "3PARdata",
 		.product       = "VV",
 		.pgpolicy      = GROUP_BY_PRIO,
@@ -225,7 +225,7 @@ static struct hwentry default_hw[] = {
 		.prio_name     = PRIO_ALUA,
 	},
 	{
-		/* Nimble Storage */
+		/* Nimble Storage / HPE Alletra 6000 */
 		.vendor        = "Nimble",
 		.product       = "Server",
 		.hwhandler     = "1 alua",
