diff -purN old/multipathd/multipathd.service new/multipathd/multipathd.service
--- old/multipathd/multipathd.service	2012-10-31 14:40:23.197092673 +0100
+++ new/multipathd/multipathd.service	2012-10-31 14:42:14.174713711 +0100
@@ -1,6 +1,6 @@
 [Unit]
 Description=Device-Mapper Multipath Device Controller
-Before=iscsi.service iscsid.service
+Before=iscsi.service iscsid.service lvm2-activation-early.service
 After=syslog.target
 ConditionPathExists=/etc/multipath.conf
 
