diff -purN old/multipathd/multipathd.service new/multipathd/multipathd.service
--- old/multipathd/multipathd.service	2012-11-02 15:03:38.024721232 +0100
+++ new/multipathd/multipathd.service	2012-11-02 15:11:54.386624285 +0100
@@ -2,6 +2,8 @@
 Description=Device-Mapper Multipath Device Controller
 Before=iscsi.service iscsid.service lvm2-activation-early.service
 After=syslog.target
+DefaultDependencies=no
+Conflicts=shutdown.target
 ConditionPathExists=/etc/multipath.conf
 
 [Service]
@@ -12,4 +14,4 @@ ExecReload=/sbin/multipathd reconfigure
 #ExecStop=/path/to/scrip delete-me if not necessary
 
 [Install]
-WantedBy=multi-user.target
+WantedBy=sysinit.target
