From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Josh Soref <2119212+jsoref@users.noreply.github.com>
Date: Sun, 31 Jul 2022 13:11:22 -0400
Subject: [PATCH] libmultipath: spelling: progress

Signed-off-by: Josh Soref <2119212+jsoref@users.noreply.github.com>
Reviewed-by: Martin Wilck <mwilck@suse.com>
Signed-off-by: Benjamin Marzinski <bmarzins@redhat.com>
---
 libmultipath/nvme/linux/nvme.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libmultipath/nvme/linux/nvme.h b/libmultipath/nvme/linux/nvme.h
index a6975549..9fc2ae7f 100644
--- a/libmultipath/nvme/linux/nvme.h
+++ b/libmultipath/nvme/linux/nvme.h
@@ -1035,7 +1035,7 @@ enum {
 	NVME_SANITIZE_LOG_STATUS_MASK		= 0x0007,
 	NVME_SANITIZE_LOG_NEVER_SANITIZED	= 0x0000,
 	NVME_SANITIZE_LOG_COMPLETED_SUCCESS	= 0x0001,
-	NVME_SANITIZE_LOG_IN_PROGESS		= 0x0002,
+	NVME_SANITIZE_LOG_IN_PROGRESS		= 0x0002,
 	NVME_SANITIZE_LOG_COMPLETED_FAILED	= 0x0003,
 	NVME_SANITIZE_LOG_ND_COMPLETED_SUCCESS	= 0x0004,
 };
