---
 libmultipath/dict.c       |   25 +++++++++++++++++++++++--
 multipathd/cli_handlers.c |    2 +-
 2 files changed, 24 insertions(+), 3 deletions(-)

Index: multipath-tools-120821/libmultipath/dict.c
===================================================================
--- multipath-tools-120821.orig/libmultipath/dict.c
+++ multipath-tools-120821/libmultipath/dict.c
@@ -2073,8 +2073,19 @@ snprint_mp_prio_args(char * buff, int le
 static int
 snprint_mp_reservation_key (char * buff, int len, void * data)
 {
+	int i;
+	unsigned char *keyp;
+	uint64_t prkey = 0;
 	struct mpentry * mpe = (struct mpentry *)data;
-	return snprintf(buff, len, "%s" , mpe->reservation_key);
+	keyp = (unsigned char *)mpe->reservation_key;
+	for (i = 0; i < 8; i++) {
+		if (i > 0)
+			prkey <<= 8;
+		prkey |= *keyp;
+		keyp++;
+	}
+			
+	return snprintf(buff, len, "0x%" PRIx64, prkey);
 }
 
 	static int
@@ -2754,7 +2765,17 @@ snprint_def_wwids_file (char * buff, int
 static int
 snprint_def_reservation_key(char * buff, int len, void * data)
 {
-	return snprintf(buff, len, "%s", conf->reservation_key);
+	int i;
+	unsigned char *keyp;
+	uint64_t prkey = 0;
+	keyp = (unsigned char *)conf->reservation_key;
+	for (i = 0; i < 8; i++) {
+		if (i > 0)
+			prkey <<= 8;
+		prkey |= *keyp;
+		keyp++;
+	}
+	return snprintf(buff, len, "0x%" PRIx64, prkey);
 }
 
 static int
Index: multipath-tools-120821/multipathd/cli_handlers.c
===================================================================
--- multipath-tools-120821.orig/multipathd/cli_handlers.c
+++ multipath-tools-120821/multipathd/cli_handlers.c
@@ -947,7 +947,7 @@ cli_getprstatus (void * v, char ** reply
 
 
 	sprintf(*reply,"%d",mpp->prflag);
-	*reply[1]='\0';
+	(*reply)[1]='\0';
 
 
 	condlog(3, "%s: reply = %s", param, *reply);
