---
 multipath/multipath.rules |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: multipath-tools-120821/multipath/multipath.rules
===================================================================
--- multipath-tools-120821.orig/multipath/multipath.rules
+++ multipath-tools-120821/multipath/multipath.rules
@@ -20,5 +20,5 @@ ENV{DM_UUID}=="mpath-?*|part[0-9]*-mpath
 ENV{DM_UUID}!="mpath-?*", GOTO="end_mpath"
 ENV{DM_SUSPENDED}=="1", GOTO="end_mpath"
 ENV{DM_ACTION}=="PATH_FAILED", GOTO="end_mpath"
-RUN+="$env{MPATH_SBIN_PATH}/kpartx -a -p p $tempnode"
+RUN+="$env{MPATH_SBIN_PATH}/kpartx -a $tempnode"
 LABEL="end_mpath"
