---
 libmultipath/dict.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: multipath-tools-120821/libmultipath/dict.c
===================================================================
--- multipath-tools-120821.orig/libmultipath/dict.c
+++ multipath-tools-120821/libmultipath/dict.c
@@ -2077,6 +2077,9 @@ snprint_mp_reservation_key (char * buff,
 	unsigned char *keyp;
 	uint64_t prkey = 0;
 	struct mpentry * mpe = (struct mpentry *)data;
+
+	if (!mpe->reservation_key)
+		return 0;
 	keyp = (unsigned char *)mpe->reservation_key;
 	for (i = 0; i < 8; i++) {
 		if (i > 0)
@@ -2768,6 +2771,9 @@ snprint_def_reservation_key(char * buff,
 	int i;
 	unsigned char *keyp;
 	uint64_t prkey = 0;
+
+	if (!conf->reservation_key)
+		return 0;
 	keyp = (unsigned char *)conf->reservation_key;
 	for (i = 0; i < 8; i++) {
 		if (i > 0)
