---
 multipathd/multipathd.service |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: multipath-tools/multipathd/multipathd.service
===================================================================
--- multipath-tools.orig/multipathd/multipathd.service
+++ multipath-tools/multipathd/multipathd.service
@@ -2,12 +2,13 @@
 Description=Device-Mapper Multipath Device Controller
 Before=iscsi.service iscsid.service
 After=syslog.target
+ConditionPathExists=/etc/multipath.conf
 
 [Service]
 Type=forking
 PIDFile=/var/run/multipathd.pid
 ExecStart=/sbin/multipathd
-ExecReload=/bin/kill -HUP $MAINPID
+ExecReload=/sbin/multipathd reconfigure
 #ExecStop=/path/to/scrip delete-me if not necessary
 
 [Install]
