---
 multipathd/main.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

Index: multipath-tools-130222/multipathd/main.c
===================================================================
--- multipath-tools-130222.orig/multipathd/main.c
+++ multipath-tools-130222/multipathd/main.c
@@ -1881,7 +1881,7 @@ main (int argc, char *argv[])
 	if (!conf)
 		exit(1);
 
-	while ((arg = getopt(argc, argv, ":dnv:k::")) != EOF ) {
+	while ((arg = getopt(argc, argv, ":dnv:k::B")) != EOF ) {
 	switch(arg) {
 		case 'd':
 			logsink = 0;
@@ -1900,6 +1900,9 @@ main (int argc, char *argv[])
 		case 'k':
 			uxclnt(optarg);
 			exit(0);
+		case 'B':
+			conf->bindings_read_only = 1;
+			break;
 		default:
 			;
 		}
