diff -uNr multipath-tools-2df6110.orig/Makefile.inc multipath-tools-2df6110/Makefile.inc
--- multipath-tools-2df6110.orig/Makefile.inc	2019-02-18 00:03:30.709950703 +0100
+++ multipath-tools-2df6110/Makefile.inc	2019-02-18 00:07:53.603837498 +0100
@@ -37,7 +37,7 @@
 
 ifndef SYSTEMD
 	ifeq ($(shell systemctl --version > /dev/null 2>&1 && echo 1), 1)
-		SYSTEMD = $(shell systemctl --version 2> /dev/null |  sed -n 's/systemd \([0-9]*\)/\1/p')
+		SYSTEMD = $(shell systemctl --version 2> /dev/null |  sed -n 's/systemd \([0-9]*\).*/\1/p')
 	endif
 endif
 
