Index: multipath-tools-temp/libmultipath/checkers/directio.c
===================================================================
--- multipath-tools-temp.orig/libmultipath/checkers/directio.c
+++ multipath-tools-temp/libmultipath/checkers/directio.c
@@ -15,6 +15,7 @@
 #include <linux/kdev_t.h>
 #include <asm/unistd.h>
 #include <libaio.h>
+#include <sys/syscall.h>
 
 #include "checkers.h"
 #include "../libmultipath/debug.h"
@@ -148,7 +149,7 @@ check_state(int fd, struct directio_cont
 	}
 	ct->running++;
 
-	r = io_getevents(ct->ioctx, 1L, 1L, &event, &timeout);
+	r = syscall(SYS_io_getevents, ct->ioctx, 1L, 1L, &event, &timeout);
 	LOG(3, "async io getevents returns %li (errno=%s)", r, strerror(errno));
 
 	if (r < 1L) {
