Index: multipath-tools-090724/libmultipath/checkers/directio.c
===================================================================
--- multipath-tools-090724.orig/libmultipath/checkers/directio.c
+++ multipath-tools-090724/libmultipath/checkers/directio.c
@@ -148,10 +148,11 @@ check_state(int fd, struct directio_cont
 	}
 	ct->running++;
 
+	errno = 0;
 	r = io_getevents(ct->ioctx, 1L, 1L, &event, &timeout);
 
 	if (r < 0 ) {
-		LOG(3, "async io getevents returned %li (errno=%s)", r,
+		LOG(3, "async io getevents returned %li (%s)", r,
 		    strerror(errno));
 		rc = PATH_UNCHECKED;
 	} else if (r < 1L) {
