Index: multipath-tools-080519/libmultipath/Makefile
===================================================================
--- multipath-tools-080519.orig/libmultipath/Makefile
+++ multipath-tools-080519/libmultipath/Makefile
@@ -26,11 +26,11 @@ $(LIBS): $(OBJS)
 	$(CC) $(SHARED_FLAGS) $(CFLAGS) -o $@ $(OBJS)
 
 install:
-	$(INSTALL_PROGRAM) -o root -g root -m 755 -d $(libdir)
-	$(INSTALL_PROGRAM) -o root -g root -m 755 $(LIBS) $(libdir)/$(LIBS)
+	$(INSTALL_PROGRAM) -m 755 -d $(DESTDIR)$(libdir)
+	$(INSTALL_PROGRAM) -m 755 $(LIBS) $(DESTDIR)$(libdir)/$(LIBS)
 
 uninstall:
-	rm -f $(libdir)/$(LIBS)
+	rm -f $(DESTDIR)$(libdir)/$(LIBS)
 
 clean:
 	rm -f core *.a *.o *.gz *.so
Index: multipath-tools-080519/libmultipath/checkers/Makefile
===================================================================
--- multipath-tools-080519.orig/libmultipath/checkers/Makefile
+++ multipath-tools-080519/libmultipath/checkers/Makefile
@@ -21,10 +21,10 @@ libcheck%.so: libsg.o %.o
 	$(CC) $(SHARED_FLAGS) -o $@ $^
 
 install:
-	$(INSTALL_PROGRAM) -o root -g root -m 755 $(LIBS) $(libdir)
+	$(INSTALL_PROGRAM) -m 755 $(LIBS) $(DESTDIR)$(libdir)
 
 uninstall:
-	rm -f $(libdir)/$(LIBS)
+	rm -f $(DESTDIR)$(libdir)/$(LIBS)
 
 clean:
 	rm -f core *.a *.o *.gz *.so
Index: multipath-tools-080519/libmultipath/prioritizers/Makefile
===================================================================
--- multipath-tools-080519.orig/libmultipath/prioritizers/Makefile
+++ multipath-tools-080519/libmultipath/prioritizers/Makefile
@@ -25,10 +25,10 @@ libprio%.so: %.o
 	$(CC) $(SHARED_FLAGS) -o $@ $^
 
 install: $(LIBS)
-	install -m 755 libprio*.so $(libdir)
+	$(INSTALL_PROGRAM) -m 755 libprio*.so $(DESTDIR)$(libdir)
 
 uninstall:
-	rm -f $(libdir)/libprio*.so
+	rm -f $(DESTDIR)$(libdir)/libprio*.so
 
 clean:
 	rm -f core *.a *.o *.gz *.so
